src/Rings/PBWAlgebraQuo.jl

```diff
@@ -57,46 +109,47 @@ end
 ####
 
 function ngens(Q::PBWAlgQuo)
-  return ngens(base_ring(Q))
+  return ngens(base_ring(Q))  # EQUIV  ngens(Q.sring)  ???
 end
 
 function gens(Q::PBWAlgQuo)
-  return elem_type(Q)[PBWAlgQuoElem(Q, x) for x in gens(base_ring(Q))]
+  return elem_type(Q)[PBWAlgQuoElem(Q, PBWAlgElem(Q.I.basering,x)) for x in gens(Q.sring)]
 end
 
 function gen(Q::PBWAlgQuo, i::Int)
-  return PBWAlgQuoElem(Q, gen(base_ring(Q), i))
+  return PBWAlgQuoElem(Q, PBWAlgElem(Q.I.basering, gen(Q.sring, i)))
 end
 
 function Base.getindex(Q::PBWAlgQuo, i::Int)
   return gen(Q, i)
 end
 
 function zero(Q::PBWAlgQuo)
-  return PBWAlgQuoElem(Q, zero(base_ring(Q)))
+  return PBWAlgQuoElem(Q, PBWAlgElem(Q.I.basering, zero(Q.sring)))
 end
 
 function one(Q::PBWAlgQuo)
-  return PBWAlgQuoElem(Q, one(base_ring(Q)))
+  return PBWAlgQuoElem(Q, PBWAlgElem(Q.I.basering, one(Q.sring)))
 end
 
-function simplify!(a::PBWAlgQuoElem)
+function simplify(a::PBWAlgQuoElem)
+  if (have_special_impl(parent(a)))  return a;  end; # short-cut for exterior algebras
```

Comment:
Idiomatic:
```suggestion
  have_special_impl(parent(a)) && return a   # short-cut for exterior algebras
```

Reply:
If this is a shortcut for exterior algebras, then does that mean `have_special_impl` really checks for exterior algebras? Then perhaps that would suggest that `is_exterior_algebra_impl` or so might be a better name for that function?